app/components-react/shared/TitleBar.tsx

```diff
@@ -30,35 +30,43 @@ export default function TitleBar(props: { windowId: string; className?: string }
   const primeTheme = /prime/.test(theme);
   const [errorState, setErrorState] = useState(false);
 
-  useEffect(lifecycle, []);
+  useEffect(() => {
+    lifecycle();
 
-  function lifecycle() {
+    return () => {
+      if (Utils.isDevMode()) {
+        ipcRenderer.removeAllListeners('unhandledErrorState');
+      }
+    };
+  }, []);
+
+  const lifecycle = useCallback(() => {
     if (Utils.isDevMode()) {
       ipcRenderer.on('unhandledErrorState', () => setErrorState(true));
     }
-  }
+  }, []);
 
-  function minimize() {
+  const minimize = useCallback(() => {
     remote.getCurrentWindow().minimize();
-  }
+  }, []);
 
-  function maximize() {
+  const maximize = useCallback(() => {
     const win = remote.getCurrentWindow();
 
     if (win.isMaximized()) {
       win.unmaximize();
     } else {
       win.maximize();
     }
-  }
+  }, []);
 
-  function close() {
+  const close = useCallback(() => {
     if (Utils.isMainWindow() && StreamingService.isStreaming) {
       if (!confirm($t('Are you sure you want to exit while live?'))) return;
     }
 
     remote.getCurrentWindow().close();
-  }
+  }, [StreamingService.isStreaming]);
```

Comment:
i think if we want to add this to the dependency array it needs to be a reactive version